### ISSUE TITLE
Menu example has an invalid a11y tree

### ISSUE THREAD

**Issue by devongovett** [tailwindlabs/headlessui#60]
The [menu example](https://codesandbox.io/s/headlessuireact-menu-example-b6xje?from-embed) has an invalid a11y tree because of the generic elements at the top ("Signed in as..."). The menu role can only contain menuitem or group children according to the ARIA spec, and testing in VoiceOver confirms that it is confused.

![image](https://user-images.githubusercontent.com/19409/95246638-9c7a1700-07c9-11eb-93f4-52b4b5554a50.png)

I'd recommend using a group for the menu items, and labelling it by the contents of the generic element (which should be aria-hidden to avoid an invalid tree). This will also improve announcement of the menu items, as the "Signed in as..." will also be announced when the first item is focused. 

Here's a fork of the sandbox: https://codesandbox.io/s/headlessuireact-menu-example-forked-ieud4?file=/src/App.js

The a11y tree looks like this with the changes applied:

![image](https://user-images.githubusercontent.com/19409/95246853-f2e75580-07c9-11eb-91cb-2ffed3d554d3.png)

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This will be fixed in a future version, thanks!

**Comment by adamwathan** [comment]
Linking PR to clarify 😄

https://github.com/tailwindlabs/headlessui/pull/228
